spec/facility_spec.rb

```diff
@@ -1,26 +1,126 @@
+  let(:cruz) { Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice}) }
+  let(:bolt) { Vehicle.new({vin: '987654321abcdefgh', year: 2019, make: 'Chevrolet', model: 'Bolt', engine: :ev}) }
+  let(:camaro) { Vehicle.new({vin: '1a2b3c4d5e6f', year: 1969, make: 'Chevrolet', model: 'Camaro', engine: :ice}) }
+  let(:registrant_1) { Registrant.new('Bruce', 18, true) }
+  let(:registrant_2) { Registrant.new('Penny', 16) }
+  let(:registrant_3) { Registrant.new('Tucker', 15) }
+
   describe '#initialize' do
     it 'can initialize' do
-      expect(@facility).to be_an_instance_of(Facility)
-      expect(@facility.name).to eq('DMV Tremont Branch')
-      expect(@facility.address).to eq('2855 Tremont Place Suite 118 Denver CO 80205')
-      expect(@facility.phone).to eq('(720) 865-4600')
-      expect(@facility.services).to eq([])
+      expect(facility_1).to be_an_instance_of(Facility)
+      expect(facility_1.name).to eq('DMV Tremont Branch')
+      expect(facility_1.address).to eq('2855 Tremont Place Suite 118 Denver CO 80205')
+      expect(facility_1.phone).to eq('(720) 865-4600')
+      expect(facility_1.services).to eq([])
+      expect(facility_1.daily_hours).to eq ("Mon, Tue, Thur, Fri  8:00 a.m.- 4:30 p.m. / Wed 8:30 a.m.-4:30 p.m.")
+      expect(facility_1.holiday_closed).to eq (nil)
     end
   end
 
   describe '#add service' do
     it 'can add available services' do
-      expect(@facility.services).to eq([])
-      @facility.add_service('New Drivers License')
-      @facility.add_service('Renew Drivers License')
-      @facility.add_service('Vehicle Registration')
-      expect(@facility.services).to eq(['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+      expect(facility_1.services).to eq([])
+      facility_1.add_service('New Drivers License')
+      facility_1.add_service('Renew Drivers License')
+      facility_1.add_service('Vehicle Registration')
+      expect(facility_1.services).to eq (['New Drivers License', 'Renew Drivers License', 'Vehicle Registration'])
+    end
+  end
+
+  describe '#collect_fees' do
+    it 'fees when register a vehicle based on plate type' do
+      facility_1.collect_fees(cruz)
+      facility_1.collect_fees(bolt)
+      facility_1.collect_fees(camaro)
+
+      expect(facility_1.collected_fees).to eq (325)
+    end
+  end
+
+  describe '#register_vehicle' do
+    it "can register a vehicle at facility_1" do
+      facility_1.add_service('Vehicle Registration')
+      facility_1.register_vehicle(cruz)
+      facility_1.register_vehicle(bolt)
+      facility_1.register_vehicle(camaro)
+
+      expect(facility_1.registered_vehicles).to eq ([cruz, bolt, camaro])
+    end
+
+    it "can't register a vehicle at facility_2" do
+      facility_2.register_vehicle(bolt)
+
+      expect(facility_2.register_vehicle(bolt)).to eq (nil)
+      expect(facility_2.registered_vehicles).to eq ([])
+    end
+  end
```

Comment:
What are the other side effects of the register_vehicle method? We want to check that all those side effects were changed as expected. We want to see that fees were collected appropriately, registration dates were set, and plate types set. I would expect all that to be tested here in the register_vehicle describe block.